pkg/engine/handlers/validation/validate_pss.go

```diff
@@ -119,15 +131,100 @@ func (h validatePssHandler) Process(
 		if len(pssChecks) == 0 && err == nil {
 			podSecurityChecks.Checks = pssChecks
 			logger.V(3).Info("policy rule is skipped due to policy exceptions", "exceptions", keys)
-			return resource, handlers.WithResponses(
-				engineapi.RuleSkip(rule.Name, engineapi.Validation, "rule is skipped due to policy exceptions "+strings.Join(keys, ", "), rule.ReportProperties).WithExceptions(matchedExceptions).WithPodSecurityChecks(podSecurityChecks),
-			)
+			return resource, engineapi.RuleSkip(rule.Name, engineapi.Validation, "rule is skipped due to policy exceptions "+strings.Join(keys, ", "), rule.ReportProperties).WithExceptions(matchedExceptions).WithPodSecurityChecks(podSecurityChecks)
 		}
 		msg := fmt.Sprintf(`Validation rule '%s' failed. It violates PodSecurity "%s:%s": %s`, rule.Name, podSecurity.Level, podSecurity.Version, pss.FormatChecksPrint(pssChecks))
-		return resource, handlers.WithResponses(
-			engineapi.RuleFail(rule.Name, engineapi.Validation, msg, rule.ReportProperties).WithPodSecurityChecks(podSecurityChecks),
+		ruleResponse := engineapi.RuleFail(rule.Name, engineapi.Validation, msg, rule.ReportProperties).WithPodSecurityChecks(podSecurityChecks)
+		allowExisitingViolations := rule.HasValidateAllowExistingViolations()
+		if engineutils.IsUpdateRequest(policyContext) && allowExisitingViolations {
+			logger.V(4).Info("is update request")
+			priorResp, err := h.validateOldObject(ctx, logger, policyContext, resource, rule, engineLoader, exceptions)
+			if err != nil {
+				return resource, engineapi.RuleError(rule.Name, engineapi.Validation, "failed to validate old object", err, rule.ReportProperties)
+			}
+
+			if ruleResponse.Status() == priorResp.Status() {
+				logger.V(3).Info("skipping modified resource as validation results have not changed", "oldResp", priorResp, "newResp", ruleResponse)
+				if ruleResponse.Status() == engineapi.RuleStatusPass {
+					return resource, ruleResponse
+				}
+				return resource, engineapi.RuleSkip(rule.Name, engineapi.Validation, "skipping modified resource as validation results have not changed", rule.ReportProperties)
```

Comment:
What will be the final rule response in this case? The reason I asked is `skip` status is used for certain scenarios:
>	// RuleStatusSkip indicates that the policy rule was not selected based on user inputs or applicability, for example
	// when preconditions are not met, or when conditional or global anchors are not satisfied.

Reply:
It will be skip, I think it should be skip in this case, because the policy did not pass the resource so it cannot be pass

what do you think?

Reply:
Makes sense! We need to call it out in the release notes, and docs, etc. Please create a tracking issue.

Reply:
https://github.com/kyverno/website/issues/1338